src/main/java/mate/academy/shop/dao/hibernate/BucketDaoHibernateImpl.java

```diff
@@ -0,0 +1,129 @@
+package mate.academy.shop.dao.hibernate;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.shop.anotation.Dao;
+import mate.academy.shop.anotation.Inject;
+import mate.academy.shop.dao.BucketDao;
+import mate.academy.shop.dao.ItemDao;
+import mate.academy.shop.dao.UserDao;
+import mate.academy.shop.factory.HibernateUtil;
+import mate.academy.shop.model.Bucket;
+import mate.academy.shop.model.Item;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class BucketDaoHibernateImpl implements BucketDao {
+    @Inject
+    private static ItemDao itemDao;
+    @Inject
+    private static UserDao userDao;
+
+    @Override
+    public Optional<Bucket> create(Bucket bucket) {
+        Long bucketId = null;
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession();) {
```

Comment:
```suggestion
        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
in all try with resources

Reply:
done